### PR TITLE
vim-plugins: add install hooks

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

- these hooks are helpful for custom nix code in VIM plugin expressions
- document `NIX_PATH` usage (see the commit message for further reference)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/vim-plugins/default.nix [hunk 1]
```diff
@@ -16,7 +16,7 @@ in
 
 # TL;DR
 # Add your plugin to ./vim-plugin-names
-# Regenerate via `nix-shell -p vimPlugins.pluginnames2nix --command "vim-plugin-names-to-nix"`
+# Regenerate via `NIX_PATH=nixpkgs=/path/to/your/local/fork nix-shell -p vimPlugins.pluginnames2nix --command "vim-plugin-names-to-nix"`
```
Comment (LnL7) [inline pkgs/misc/vim-plugins/default.nix]: I think `nix-shell -I nixpkgs=/path/to/nixpkgs ...` also works

**Review by LnL7 — APPROVED** [review]

**Comment by Ma27** [comment]
@LnL7 done :)
